src/internal/backoff/exponentialbackoff.go

```diff
@@ -0,0 +1,45 @@
+package backoff
+
+import "time"
+
+// ExponentialBackoff is a struct that handles backoff duration.
+type ExponentialBackoff struct {
+	initialBackoff       time.Duration
+	repeatedBackoff      time.Duration
+	finalBackoff         time.Duration
+	initialFailureCount  int
+	repeatedFailureCount int
+	failures             int
+}
+
+// NewExponentialBackoff creates ExponentialBackoff struct.
+func NewExponentialBackoff(initialBackoff time.Duration, repeatedBackoff time.Duration, finalBackoff time.Duration, initialFailureCount int, repeatedFailureCount int) *ExponentialBackoff {
```

Comment:
I guess we can wrap those params in a struct. If there are no circular dependencies, you can reuse it in `Config` as well.

Reply:
nah, It'd bring too much complexity for such a simple functionality and It'd allow to omit mandatory values

Reply:
You can still omit them - it's a matter of validation actually.

Reply:
`NewExponentialBackoff(0, 0, 0, 0, 0)` is more verbose than 
`NewExponentialBackoff(ExponentialBackoffOptions{})`
`NewExponentialBackoff()`
or
`NewExponentialBackoff(nil)`

Reply:
Yes, we can - it's a matter of taste. I just pointed out that making the arguments verbose does not protect you from providing invalid values to the constructor.